src/cryptonote_core/blockchain.h

```diff
@@ -1581,7 +1581,7 @@ namespace cryptonote
      * can be reconstituted by the receiver. This function expands
      * that implicit data.
      */
-    bool expand_transaction_2(transaction &tx, const crypto::hash &tx_prefix_hash, const std::vector<std::vector<rct::ctkey>> &pubkeys) const;
+    static bool expand_transaction_2(transaction &tx, const crypto::hash &tx_prefix_hash, const std::vector<std::vector<rct::ctkey>> &pubkeys);
```

Comment:
Seems strange to have a static function in a header file

Reply:
Should be a static method of `class Blockchain`, no?

Reply:
See https://github.com/monero-project/monero/pull/8707#issuecomment-1385838246, the various declaration changes suggest an ownership/scoping problem which should be fixed in this PR.